lib/trino-filesystem/src/main/java/io/trino/filesystem/memory/MemoryInputStream.java

```diff
@@ -78,6 +79,7 @@ public int read(byte[] destination, int destinationIndex, int length)
             throws IOException
     {
         ensureOpen();
+        checkFromIndexSize(destinationIndex, length, destination.length);
```

Comment:
I believe this is already validated by the underlying SliceInput. Drop this commit

Reply:
This is required for `MemoryInputStream` to implement `InputStream` correctly. The various file system `InputStream` implementors currently have inconsistent behaviour, see https://github.com/trinodb/trino/pull/19813

The only use for it at the moment is to make the fuzz tests pass, so we could potentially drop it.

Reply:
I suspect that we hit https://github.com/airlift/slice/blob/fec82db84a122d294f5086deab2102ce2b2597d6/src/main/java/io/airlift/slice/BasicSliceInput.java#L174
and return -1 instead of throwing an exception for `inputStream.read(new byte[1], -1, 0)`

Reply:
Okay, let's try to land https://github.com/trinodb/trino/pull/19813 first then